ocaml/testsuite/tests/regression/pr2098/stub.c

```diff
@@ -0,0 +1,23 @@
+#include <caml/custom.h>
+
+static void caml_test_finalize(value v)
+{
+  if (Caml_state_field(in_minor_collection))
+    caml_fatal_error("Thread switch from inside minor GC");
```

Comment:
I was expecting this to be something like:
```
caml_enter_blocking_section();
caml_leave_blocking_section();
```
so that we test the error message from the runtime, instead of duplicating it in a test. I see one of the other versions of this file worked like this. Did you work out what the flakiness was?